File: system/fake/FakeGraphics.java
package common.system.fake;

public interface FakeGraphics {

	int RED = 0, YELLOW = 1, BLACK = 2, MAGENTA = 3, BLUE = 4, CYAN = 5, WHITE = 6;
	int DEF = 0, TRANS = 1, BLEND = 2, GRAY = 3;

	void colRect(float x, float y, float w, float h, int r, int g, int b, int a);

	default void delete(FakeTransform at) {

	}

	void drawImage(FakeImage bimg, float x, float y);

	void drawImage(FakeImage bimg, float x, float y, float d, float e);

	void drawLine(float i, float j, float x, float y);

	void drawOval(float i, float j, float k, float l);

	void drawRect(float x, float y, float w, float h);

	void fillOval(float i, float j, float k, float l);

	void fillRect(float x, float y, float w, float h);

	FakeTransform getTransform();

	void gradRect(float x, float y, float w, float h, float a, float b, int[] c, float d, float e, int[] f);

	void gradRectAlpha(float x, float y, float w, float h, float a, float b, int al, int[] c, float d, float e, int al2, int[] f);

	void rotate(float d);

	void scale(float hf, float vf);

	void setColor(int c);

	void setColor(int r, int g, int b);

	void setComposite(int mode, int p0, int p1);

	void setRenderingHint(int key, int object);

	void setTransform(FakeTransform at);

	void translate(float x, float y);

}

File: system/fake/FakeGraphics.kt
package common.system.fake

interface FakeGraphics {
    fun colRect(x: Int, y: Int, w: Int, h: Int, r: Int, g: Int, b: Int, a: Int)
    fun delete(at: FakeTransform?) {}
    fun drawImage(bimg: FakeImage, x: Double, y: Double)
    fun drawImage(bimg: FakeImage, x: Double, y: Double, d: Double, e: Double)
    fun drawLine(i: Int, j: Int, x: Int, y: Int)
    fun drawOval(i: Int, j: Int, k: Int, l: Int)
    fun drawRect(x: Int, y: Int, x2: Int, y2: Int)
    fun fillOval(i: Int, j: Int, k: Int, l: Int)
    fun fillRect(x: Int, y: Int, w: Int, h: Int)
    fun getTransform(): FakeTransform
    fun gradRect(x: Int, y: Int, w: Int, h: Int, a: Int, b: Int, c: IntArray, d: Int, e: Int, f: IntArray)
    fun rotate(d: Double)
    fun scale(hf: Int, vf: Int)
    fun setColor(c: Int)
    fun setComposite(mode: Int, p0: Int, p1: Int)
    fun setRenderingHint(key: Int, `object`: Int)
    fun setTransform(at: FakeTransform)
    fun translate(x: Double, y: Double)

    companion object {
        const val RED = 0
        const val YELLOW = 1
        const val BLACK = 2
        const val MAGENTA = 3
        const val BLUE = 4
        const val CYAN = 5
        const val WHITE = 6
        const val DEF = 0
        const val TRANS = 1
        const val BLEND = 2
        const val GRAY = 3
    }
}
